src/libs/actions/IOU.js

```diff
@@ -722,11 +723,219 @@ function payIOUReport({
     return promiseWithHandlers;
 }
 
+/**
+ * @param {String} chatReportID - Report ID of the chat where the IOU is.
+ * @param {Number} amount - IOU amount in cents.
+ * @param {String} currency - IOU currency.
+ * @param {String} comment
+ * @param {Array} participants - Participants on the IOU
+ */
+function sendMoneyWithWallet(chatReportID, amount, currency, comment, participants) {
+    const chatReport = _.empty(chatReportID) ? chatReports[`${ONYXKEYS.COLLECTION.REPORT}${chatReportID}`] : ReportUtils.buildOptimisticChatReport(participants);
+    const ownerEmail = participants[0].login;
+    const userEmail = participants[1].login;
+    const optimisticIOUReport = ReportUtils.buildOptimisticIOUReport(
+        ownerEmail,
+        userEmail,
+        amount,
+        chatReport.chatReportID,
+        currency,
+        preferredLocale,
+    );
+    const newSequenceNumber = Report.getMaxSequenceNumber(chatReport.reportID) + 1;
+    const optimisticIOUReportAction = ReportUtils.buildOptimisticIOUReportAction(
+        newSequenceNumber,
+        CONST.IOU.REPORT_ACTION_TYPE.PAY,
+        amount,
+        currency,
+        comment,
+        participants,
+        CONST.IOU.PAYMENT_TYPE.EXPENSIFY,
+        '',
+        optimisticIOUReport.reportID,
+    );
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${chatReport.chatReportID}`,
+            value: {
+                [optimisticIOUReportAction.reportActionID]: {
+                    optimisticIOUReportAction,
+
+                    // Set pendingAction state for pattern B
+                    pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                },
+            },
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${chatReport.chatReportID}`,
+            value: chatReport.chatReportID,
```

Comment:
If we are only putting the chatReportID in here, but the report didn't exist before (we created an optimistic one), then the user won't be able to see it while offline right?

Reply:
+1

Reply:
@sketchydroide In case of chat report very much same as [in requestMoney](https://github.com/Expensify/App/blob/377dbd9b05e185c359c06c6269c671093538fa28/src/libs/actions/IOU.js#L92-L107)

Basically we need to assign the same values of the chat report, in case it was generated chat report, we need to use Set onyx update method.

I think we should update this flow same as there, ie make `isNewChat` variable and then change the method to be SET

Reply:
thanks, done I think